<html lang="en">

<head>
    <meta charset="utf-8">
    <title>PGS Catalog - Download Information</title>
    <meta name="keywords"
        content="polygenic score, polygenic risk score, pgs, prs, catalog, catalogue, database, genetic, genomic, prediction, disease, gwas, ebi">
    <!-- 3 to 10 keywords about the content of this page (not the whole project) -->
    <meta name="author" content="PGS Catalog">
    <meta name="description" content="The Polygenic Score (PGS) Catalog is an open database of
      published PGS and the relevant metadata needed to apply and evaluate them correctly.">
    <meta name="HandheldFriendly" content="true">
    <meta name="MobileOptimized" content="width">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <meta name="theme-color" content="#007C82"> <!-- Android Chrome mobile browser tab color -->
    <!-- Get suggested SEO and social metatags at:
         https://www.ebi.ac.uk/style-lab/websites/patterns/meta-copy.html -->

    <!-- Add information on the life cycle of this page -->
    <meta name="ebi:owner" content="Samuel Lambert"> <!-- Who should be contacted about changes -->
    <meta name="ebi:review-cycle" content="30"> <!-- In days, how often should the content be reviewed -->
    <meta name="ebi:last-review" content="2019-07-25"> <!-- The last time the content was reviewed -->
    <meta name="ebi:expiry" content="2019-12-31"> <!-- When this content is no longer relevant -->

    <!-- If you link to any other sites frequently, consider optimising performance with a DNS prefetch -->
    <link rel="dns-prefetch" href="//www.ebi.ac.uk/gwas/">

    <!-- If you have custom icon, replace these as appropriate.
         You can generate them at realfavicongenerator.net -->
    <link rel="icon" type="image/x-icon" href="/static/catalog/imgs/favicons/favicon.ico">
    <link rel="icon" type="image/png" href="/static/catalog/imgs/favicons/favicon-16x16.png">
    <link rel="icon" type="image/png" href="/static/catalog/imgs/favicons/favicon-32x32.png">
    <link rel="icon" type="image/png" href="/static/catalog/imgs/favicons/apple-touch-icon.png">
    <link rel="icon" type="image/png" sizes="192x192" href="/static/catalog/imgs/favicons/android-chrome-192x192.png">
    <!-- Android (192px) -->
    <link rel="icon" type="image/png" sizes="384x384" href="/static/catalog/imgs/favicons/android-chrome-384x384.png">
    <!-- Android (384px) -->
    <meta name="msapplication-TileImage" content="/static/catalog/imgs/favicons/mstile-150x150.png">

    <!-- (S)CSS libraries -->



    <!-- Boostrap CSS libraries -->
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.2/dist/css/bootstrap.min.css"
        integrity="sha384-xOolHFLEh07PJGoPkLv1IbcEPTNtaed2xpHsD9ESMhqIYd0nLMwNLD69Npy4HI+N" crossorigin="anonymous">
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v6.3.0/css/all.css" crossorigin="anonymous">


    <!-- jQuery UI -->
    <link rel="stylesheet" href="//code.jquery.com/ui/1.13.2/themes/base/jquery-ui.css">

    <!-- CSS: implied media=all -->
    <!-- CSS concatenated and minified via ant build script -->


    <!-- PGS (S)CSS file -->

    <link rel="stylesheet" href="/static/catalog/pgs.min.css" type="text/css" media="all">






    <!-- Javascript libraries -->



    <!-- jQuery -->
    <script src="https://code.jquery.com/jquery-3.7.1.min.js"
        integrity="sha256-/JqT3SQfawRcv/BIHPThkBvs0OEvtFFmqPF/lYI/Cxo=" crossorigin="anonymous"></script>
    <!-- jQuery UI -->
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jqueryui/1.13.2/jquery-ui.min.js"></script>


    <!-- Use of GWAS API - Javascript file -->


    <!-- Use of Release JS - Javascript file -->


    <!-- Bootstrap bundle include Popper.js and Bootstrap JS -->
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.2/dist/js/bootstrap.bundle.min.js"
        integrity="sha384-Fy6S3B9q64WdZWQUiU+q4/2Lc9npb8tCaSX9FK7E8HnRr0Jz8D6OP9dO5Vg3Q9ct"
        crossorigin="anonymous"></script>


    <!-- D3 graph library -->


    <!-- Chart.js -->


    <!-- PGS Javascript file -->

    <script src="/static/catalog/pgs.min.js"></script>







    <script
        src="https://ebi.emblstatic.net/web_guidelines/EBI-Framework/v1.3/js/ebi-global-includes/script/5_ebiFrameworkNotificationBanner.js?legacyRequest=1.3"></script>
</head>

<body style="padding-bottom: 67px;">


    <div>
        <nav class="navbar fixed-top navbar-expand-lg">
            <button type="button" class="navbar-toggler" data-toggle="collapse" data-target="#navbarCollapse">
                <span class="navbar-toggler-icon"></span>
            </button>


            <a class="navbar-brand" href="/" data-original-title="" title="Polygenic Score Catalog">
                <img src="/static/catalog/imgs/PGS_Logo.png"><span>PGS Catalog</span>
            </a>


            <!--/.navbar-collapse - begin -->

            <!-- Horizontal menu -->
            <div class="collapse navbar-collapse" id="navbarCollapse">
                <ul class="nav navbar-nav">

                    <li class="nav-item active"><a class="nav-link" href="/">Home</a></li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Browse</a>
                        <div class="dropdown-menu">
                            <a class="dropdown-item" href="/browse/scores/">Polygenic Scores</a>
                            <a class="dropdown-item" href="/browse/traits/">Traits</a>
                            <a class="dropdown-item" href="/browse/studies/">Publications</a>
                        </div>
                    </li>
                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Downloads</a>
                        <div class="dropdown-menu">
                            <a class="dropdown-item" href="/downloads/">Data Downloads</a>
                            <a class="dropdown-item" href="/rest/">REST API</a>
                        </div>
                    </li>
                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Documentation</a>
                        <div class="dropdown-menu">
                            <a class="dropdown-item" href="/about/">About</a>
                            <a class="dropdown-item" href="/docs/ancestry/">Ancestry Data</a>
                            <a class="dropdown-item" href="/docs/">Data Description</a>
                            <a class="dropdown-item" href="/docs/faq">FAQ</a>

                        </div>
                    </li>
                </ul>
            </div>
            <!--/.navbar-collapse - end -->

            <!-- Search box -->
            <div class="search_box">

                <form class="input-group search_form" id="search_form" action="/search" method="get">
                    <input class="form-control ui-autocomplete-input" id="q" name="q" type="text"
                        placeholder="Search..." spellcheck="false" autocomplete="off">
                    <div class="input-group-append">
                        <button type="button" class="btn btn-pgs pgs_no_icon_link" id="search_btn" value="Submit">
                            <span class="fas fa-search"></span>
                        </button>
                    </div>
                </form>
                <div class="search_examples">
                    <a href="/search?q=breast cancer">breast cancer</a>, <a href="/search?q=glaucoma">glaucoma</a>, <a
                        href="/search?q=BMI">BMI</a><span class="extra_example">, <a
                            href="/search?q=EFO_0001645">EFO_0001645</a></span>
                </div>

            </div>
        </nav>
    </div>

    <div class="search_box_2" style="display:none">
        <form class="input-group search_form" id="search_form_2" action="/search" method="get">
            <input class="form-control ui-autocomplete-input" id="q2" name="q" type="text" placeholder="Search..."
                spellcheck="false" autocomplete="off">
            <div class="input-group-append">
                <button type="button" class="btn btn-pgs pgs_no_icon_link" id="search_btn_2" value="Submit">
                    <span class="fas fa-search"></span>
                </button>
            </div>
        </form>
        <div class="search_examples">
            Examples: <a href="/search?q=breast cancer">breast cancer</a>, <a href="/search?q=glaucoma">glaucoma</a>, <a
                href="/search?q=BMI">BMI</a><span class="extra_example">, <a
                    href="/search?q=EFO_0001645">EFO_0001645</a></span>
        </div>
    </div>


    <div class="feedback">
        <a href="mailto:pgs-info@ebi.ac.uk" title="Feedback or contact us">
            <button type="button" id="feedback-btn">Feedback</button>
        </a>
    </div>

    <div class="container-fluid container-extra min-vh-100">


        <nav aria-label="You are here:">
            <ol class="breadcrumb">
                <li class="breadcrumb-item"><a href="/">PGS Catalog</a></li>
                <li class="breadcrumb-item">Downloads</li>
            </ol>
        </nav>


        <h2 class="mr-auto">Downloads</h2>

        <div class="content_container">
            <div class="table-of-contents">
                <nav role="navigation">
                    <h5>Table of Contents</h5>
                    <ul>
                        <li><a href="#dl_ftp_list">Available PGS Catalog downloads</a></li>
                        <li><a href="#dl_ftp">PGS Catalog FTP structure</a></li>
                        <li><a href="#dl_ftp_metadata">PGS Catalog Metadata</a></li>
                        <li>
                            <a href="#dl_scoring_files">PGS Scoring Files</a>
                            <ul>
                                <li>
                                    <a href="#dl_ftp_scoring">Formatted Files</a>
                                    <ul>
                                        <li><i class="fas fa-hashtag pgs_color_1"></i> <a
                                                href="#scoring_header">Header</a><span class="pl-2 pr-2">—</span><i
                                                class="fas fa-th pgs_color_1"></i> <a
                                                href="#scoring_columns">Columns</a><span class="pl-2 pr-2">—</span><i
                                                class="fas fa-copy pgs_color_1"></i> <a href="#scoring_changes">Format
                                                changes</a><span class="pl-2 pr-2">—</span><i
                                                class="far fa-copy pgs_color_1"></i> <a
                                                href="#scoring_previous">Previous formats</a></li>
                                    </ul>
                                </li>
                                <li>
                                    <a href="#dl_ftp_scoring_hm_pos">Harmonized Files</a>
                                    <ul>
                                        <li><i class="fas fa-file-signature pgs_color_1"></i> <a href="#hm_pos_fn">File
                                                name</a><span class="pl-2 pr-2">—</span><i
                                                class="fas fa-hashtag pgs_color_1"></i> <a
                                                href="#hm_pos_header">Header</a><span class="pl-2 pr-2">—</span><i
                                                class="fas fa-th pgs_color_1"></i> <a href="#hm_pos_columns">Additional
                                                Columns</a></li>
                                    </ul>
                                </li>
                            </ul>
                        </li>

                    </ul>
                </nav>
            </div>
        </div>


        <section>
            <h3 id="dl_ftp_list">Available PGS Catalog downloads</h3>
            <table class="table table-bordered table_pgs_h">
                <tbody>
                    <tr>
                        <td>
                            <span class="pgs_h_big_header">PGS Scoring Files &amp; Metadata</span>
                            <div class="mt-1">
                                <small>Individual PGS variants scoring and metadata files</small>
                            </div>
                        </td>
                        <td><a class="btn btn-pgs pgs_no_icon_link pgs-ftp-btn"
                                href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/" target="_blank"><span
                                    class="fas fa-folder"></span> View PGS Score Directories (FTP)</a></td>
                    </tr>
                    <tr>
                        <td>
                            <span class="pgs_h_big_header">PGS Catalog Metadata</span>
                            <div class="mt-1">
                                <small>Available PGS global metadata files</small>
                            </div>
                        </td>
                        <td><a class="btn btn-pgs pgs_no_icon_link pgs-ftp-btn"
                                href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/metadata/" target="_blank"><span
                                    class="fas fa-folder"></span> Bulk Metadata Downloads (FTP)</a></td>
                    </tr>
                    <tr>
                        <td>
                            <span class="pgs_h_big_header">PGS Catalog REST API</span>
                            <div class="mt-1">
                                <small>Programmatic access to the PGS Catalog metadata</small>
                            </div>
                        </td>
                        <td><a class="btn btn-pgs pgs_no_icon_link" href="/rest/"><span class="fas fa-code"></span> REST
                                API endpoint documentation</a></td>
                    </tr>
                    <tr>
                        <td>
                            <span class="pgs_h_big_header">Python package <span
                                    class="pgs_color_1">pgscatalog_utils</span></span>
                            <div class="mt-1">
                                <small>A collection of tools, such as scoring files download</small>
                            </div>
                        </td>
                        <td><a class="btn btn-pgs pgs_no_icon_link" href="https://pypi.org/project/pgscatalog-utils/"
                                target="_blank"><span class="fa-brands fa-python"></span> Python package
                                documentation</a></td>
                    </tr>
                </tbody>
            </table>
        </section>


        <section>
            <h3 id="dl_ftp">PGS Catalog FTP structure</h3>
            <p>The PGS Catalog FTP allows for consistent access to the bulk downloads, and is indexed by Polygenic Score
                (PGS) ID to allow programmatic access to score level data. The following diagram illustrates the FTP
                structure:</p>

            <div class="pgs_formatted_block">
                <pre><a href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs" target="_blank" class="external-link">ftp://ftp.ebi.ac.uk/pub/databases/spot/pgs</a>
  ├── <span class="pgs_hl">pgs_scores_list.txt</span> <small>(list of Polygenic Score IDs)</small>
  ├── <a href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/metadata/" target="_blank" class="external-link"><b>metadata/</b></a>
  │&nbsp;&nbsp;   ├── <span class="pgs_hl">pgs_all_metadata.xlsx</span>
  │&nbsp;&nbsp;   ├── <span class="pgs_hl">pgs_all_metadata_</span>[<span class="pgs_hl_i">sheet_name</span>]<span class="pgs_hl">.csv</span> <small>(7 files)</small>
  │&nbsp;&nbsp;   ├── <span class="pgs_hl">pgs_all_metadata.tar.gz</span> <small>(xlsx + csv files)</small>
  │&nbsp;&nbsp;   ├── <a href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/metadata//publications/" target="_blank" class="external-link"><b>publications/</b></a> <small>(metadata for large studies)</small>
  │&nbsp;&nbsp;   └── <a href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/metadata//previous_releases/" target="_blank" class="external-link"><b>previous_releases/</b></a>
  └── <a href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores" target="_blank" class="external-link"><b>scores/</b></a>
        ├── <a href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000001/" target="_blank" class="external-link"><b>PGS000001/</b></a>
        │&nbsp;&nbsp;   ├── <a href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000001/Metadata/" target="_blank" class="external-link"><b>Metadata/</b></a>
        │&nbsp;&nbsp;   │&nbsp;&nbsp;   ├── <span class="pgs_hl">PGS000001_metadata.xlsx</span>
        │&nbsp;&nbsp;   │&nbsp;&nbsp;   ├── <span class="pgs_hl">PGS000001_metadata_</span>[<span class="pgs_hl_i">sheet_name</span>]<span class="pgs_hl">.csv</span> <small>(7 files)</small>
        │&nbsp;&nbsp;   │&nbsp;&nbsp;   ├── <span class="pgs_hl">PGS000001_metadata.tar.gz</span> <small>(xlsx + csv files)</small>
        │&nbsp;&nbsp;   │&nbsp;&nbsp;   └── <b>archived_versions/</b>
        │&nbsp;&nbsp;   └── <a href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000001/ScoringFiles" target="_blank" class="external-link"><b>ScoringFiles/</b></a>
        │&nbsp;&nbsp;         ├── <span class="pgs_hl">PGS000001.txt.gz</span>
        │&nbsp;&nbsp;         ├── <b>archived_versions/</b>
        │&nbsp;&nbsp;         └── <a href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000001/ScoringFiles/Harmonized" target="_blank" class="external-link"><b>Harmonized/</b></a>
        │                 ├── <span class="pgs_hl">PGS000001_hmPOS_GRCh37.txt.gz</span>
        │&nbsp;&nbsp;&nbsp;              └── <span class="pgs_hl">PGS000001_hmPOS_GRCh38.txt.gz</span>
        ├── <a href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000002/" target="_blank" class="external-link"><b>PGS000002/</b></a>
        ·     ├─ ···
        ·     └─ ···
        ·
        └── <b>PGS00XXXX/</b>
              ├─ ···
              └─ ···</pre>
            </div>
        </section>

        <section>
            <h3 class="mt-5" id="dl_ftp_metadata">PGS Catalog Metadata</h3>

            <div class="ml-3">
                <p>Bulk download of the entire PGS Catalog's metadata, describing all PGS in terms of their publication
                    source, samples used for development/evaluation, and related performance metrics.<a
                        class="btn btn-pgs pgs_no_icon_link ml-2"
                        href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/metadata/pgs_all_metadata.xlsx"
                        target="_blank" download=""><span class="fas fa-download"></span> Download Metadata file<span
                            class="badge badge-pgs-white ml-2">.xlsx</span></a></p>
                <p>The bulk download contains a single Excel file with multiple sheets describing each of the data
                    types. The sheets are also provided as individual <code>.csv</code> files for easier import in
                    analysis tools, and are provided on the FTP in the <code>metadata/</code> folder.</p>
                <div class="table-responsive">
                    <table class="table table-striped table_pgs_auto">
                        <thead class="thead_pgs">
                            <tr>
                                <th>Worksheet</th>
                                <th>Description</th>
                                <th><span class="fas fa-download"></span> Download Sheet<span
                                        class="badge badge-pgs-white ml-2">.csv</span></th>
                            </tr>
                        </thead>
                        <tbody>
                            <tr>
                                <td>Readme</td>
                                <td>PGS Catalog Release Date and Summary Information.</td>
                                <td>-</td>
                            </tr>
                            <tr>
                                <td>Publications</td>
                                <td>Lists the publication sources for the PGS and PGS evaluations in the catalog.</td>
                                <td><a class="pgs_no_icon_link file_link"
                                        href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/metadata/pgs_all_metadata_publications.csv"
                                        title="Click to download all PGS Publications" download="" target="_blank"></a>
                                </td>
                            </tr>
                            <tr>
                                <td>EFO Traits</td>
                                <td>Lists the ontology-mapped traits information for all PGS in the catalog.</td>
                                <td><a class="pgs_no_icon_link file_link"
                                        href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/metadata/pgs_all_metadata_efo_traits.csv"
                                        title="Click to download all PGS Traits" download="" target="_blank"></a></td>
                            </tr>
                            <tr>
                                <td>Scores</td>
                                <td>Lists all PGS scores and their associated metadata.</td>
                                <td><a class="pgs_no_icon_link file_link pgs_helptip"
                                        href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/metadata/pgs_all_metadata_scores.csv"
                                        title="" download="" target="_blank" data-toggle="tooltip"
                                        data-placement="bottom" data-delay="800"
                                        data-original-title="Click to download all PGS Score metadata"></a></td>
                            </tr>
                            <tr>
                                <td>Score Development Samples</td>
                                <td>Lists the samples used to create the PGS: samples used to discover the variant
                                    associations (GWAS), samples used for score development/training.</td>
                                <td><a class="pgs_no_icon_link file_link"
                                        href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/metadata/pgs_all_metadata_score_development_samples.csv"
                                        title="Click to download all PGS Score development samples" download=""
                                        target="_blank"></a></td>
                            </tr>
                            <tr>
                                <td>Performance Metrics</td>
                                <td>Lists all performance metrics and the associated PGS Scores and Publications.</td>
                                <td><a class="pgs_no_icon_link file_link"
                                        href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/metadata/pgs_all_metadata_performance_metrics.csv"
                                        title="Click to download all PGS Performance metrics" download=""
                                        target="_blank"></a></td>
                            </tr>
                            <tr>
                                <td>Evaluation Sample Sets</td>
                                <td>Describes the samples used to evaluate PGS performance (refferenced as Polygenic
                                    Score Sample Sets (PSS).</td>
                                <td><a class="pgs_no_icon_link file_link"
                                        href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/metadata/pgs_all_metadata_evaluation_sample_sets.csv"
                                        title="Click to download all PGS Evaluation sample sets" download=""
                                        target="_blank"></a></td>
                            </tr>
                            <tr>
                                <td>Cohorts</td>
                                <td>Lists all the cohorts used in the different samples.</td>
                                <td><a class="pgs_no_icon_link file_link pgs_helptip"
                                        href="https://ftp.ebi.ac.uk/pub/databases/spot/pgs/metadata/pgs_all_metadata_cohorts.csv"
                                        title="" download="" target="_blank" data-toggle="tooltip"
                                        data-placement="bottom" data-delay="800"
                                        data-original-title="Click to download all PGS Cohorts metadata"></a></td>
                            </tr>
                        </tbody>
                    </table>
                </div>
            </div>

        </section>

        <section>
            <h3 class="mr-3" id="dl_scoring_files">PGS Scoring Files</h3>
        </section>

        <section>
            <div class="d-inline-flex">
                <h4 class="mr-3" id="dl_ftp_scoring">Formatted Files</h4>
                <div class="mt-4">
                    <div class="mt-2 badge badge-pgs">Format: 2.0</div>
                </div>
            </div>

            <div class="ml-3">
                <div class="d-flex mb-3">
                    <div class="pgs_shaded_block">
                        <i class="fas fa-hashtag pgs_color_1"></i> <a href="#scoring_header">Header</a><span
                            class="pl-2 pr-2">—</span><i class="fas fa-th pgs_color_1"></i> <a
                            href="#scoring_columns">Columns</a><span class="pl-2 pr-2">—</span><i
                            class="fas fa-copy pgs_color_1"></i> <a href="#scoring_changes">Format changes</a><span
                            class="pl-2 pr-2">—</span><i class="far fa-copy pgs_color_1"></i> <a
                            href="#scoring_previous">Previous formats</a>
                    </div>
                </div>
                <p>
                    Each scoring file (variant information, effect alleles/weights) is formatted to be a gzipped
                    <b>tab-delimited</b> text file, labelled by its PGS Catalog Score ID (e.g.
                    <code>PGS000001.txt.gz</code>).
                </p>
                They can be found in URLs like this:
                <div class="pgs_formatted_block">
                    <pre>ftp://ftp.ebi.ac.uk/pub/databases/spot/pgs/<b>scores</b>/<b>PGS######</b>/<b>ScoringFiles</b>/</pre>
                </div>

                <div class="d-flex">
                    <div class="mb-2 pgs_note pgs_note_1">
                        <div>
                            <span>Note: </span>These files are composed of author-reported variants and annotations, and
                            have only been consistently formatted to have the same column headings and data types within
                            each column. We are currently working on methods to provide harmonized versions of each PGS
                            in different genome builds (GRCh37 and GRCh38), ensuring each variant has a chromosomal
                            position (either by adding it based on rsID or using liftover), and flagging potentially
                            problematic variants (e.g. palindromic SNPs) or those that are inconsistent with the current
                            genome build (e.g. strand flips, and variants not found in the Ensembl Variation databases).
                        </div>
                    </div>
                </div>

                <h5 id="scoring_header" class="mt-4"><i class="fas fa-hashtag pgs_color_1 mr-2"></i>Formatted Files
                    <span class="pgs_color_1">—</span> Header</h5>
                <p>Here is a description of the PGS Scoring Files header:</p>

                <div class="pgs_formatted_block">
                    <pre><span>###PGS CATALOG SCORING FILE - see https://www.pgscatalog.org/downloads/#dl_ftp_scoring for additional information</span>
<b>#format_version</b>=<i>Version of the scoring file format, e.g. '2.0'</i>
<span>##POLYGENIC SCORE (PGS) INFORMATION</span>
<b>#pgs_id</b>=<i>PGS identifier, e.g. 'PGS000001'</i>
<b>#pgs_name</b>=<i>PGS name, e.g. 'PRS77_BC' - optional</i>
<b>#trait_reported</b>=<i>trait, e.g. 'Breast Cancer'</i>
<b>#trait_mapped</b>=<i>Ontology trait name, e.g. 'breast carcinoma'</i>
<b>#trait_efo</b>=<i>Ontology trait ID (EFO), e.g. 'EFO_0000305'</i>
<b>#genome_build</b>=<i>Genome build/assembly, e.g. 'GRCh38'</i>
<b>#variants_number</b>=<i>Number of variants listed in the PGS</i>
<b>#weight_type</b>=<i>Variant weight type, e.g. 'beta', 'OR/HR' (default 'NR')</i>
<span>##SOURCE INFORMATION</span>
<b>#pgp_id</b>=<i>PGS publication identifier, e.g. 'PGP000001'</i>
<b>#citation</b>=<i>Information about the publication</i>
<b>#license</b>=<i>License and terms of PGS use/distribution - refers to the EMBL-EBI Terms of Use by default</i>
<span class="scoring_col">rsID</span><span class="scoring_col">chr_name</span><span class="scoring_col">chr_position</span><span class="scoring_col">effect_allele</span><span class="scoring_col">other_allele</span><span class="scoring_col">effect_weight</span><span class="pgs_color_2">...</span></pre>
                </div>
                <div class="d-flex mb-1">
                    <div class="d-flex mb-2 pgs_note pgs_note_1">
                        <div>
                            <span>Note: </span>The "<b>#trait_mapped</b>" and "<b>#trait_efo</b>" metadata can be
                            composed of more than one value (separated by a <code class="pgs_color_2">|</code> "pipe"),
                            e.g.:
                            <pre class="mt-1 mb-0"><b>#trait_mapped</b>=Ischemic stroke<span class="pgs_color_2">|</span>stroke
<b>#trait_efo</b>=HP_0002140<span class="pgs_color_2">|</span>EFO_0000712</pre>
                        </div>
                    </div>
                </div>

                <i class="fas fa-angle-double-right pgs_color_1"></i> <a class="toggle_btn pgs_btn_plus"
                    data-toggle="tooltip" data-placement="right" data-delay="500" id="pgs_header_example" title=""
                    data-original-title="Click to show/hide an example of Scoring Files header">Example of PGS Scoring
                    Files header</a>
                <div class="toggle_list mt-2" id="list_pgs_header_example">
                    <div class="pgs_formatted_block">
                        <pre
                            class="example"><span>###PGS CATALOG SCORING FILE - see https://www.pgscatalog.org/downloads/#dl_ftp_scoring for additional information</span>
<b>#format_version</b>=2.0
<span>##POLYGENIC SCORE (PGS) INFORMATION</span>
<b>#pgs_id</b>=PGS000348
<b>#pgs_name</b>=PRS_PrCa
<b>#trait_reported</b>=Prostate cancer
<b>#trait_mapped</b>=prostate carcinoma
<b>#trait_efo</b>=EFO_0001663
<b>#genome_build</b>=GRCh37
<b>#variants_number</b>=72
<b>#weight_type</b>=log(OR)
<span>##SOURCE INFORMATION</span>
<b>#pgp_id</b>=PGP000113
<b>#citation</b>=Black M et al. Prostate (2020). doi:10.1002/pros.24058
<b>#license</b>=Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International (CC BY-NC-ND 4.0). © 2020 Ambry Genetics.
<span class="scoring_col">rsID</span><span class="scoring_col">chr_name</span><span class="scoring_col">chr_position</span><span class="scoring_col">effect_allele</span><span class="scoring_col">other_allele</span><span class="scoring_col">effect_weight</span><span class="pgs_color_2">...</span></pre>
                    </div>
                </div>

                <h5 id="scoring_columns" class="mt-5"><i class="fas fa-th pgs_color_1 mr-2"></i>Formatted Files <span
                        class="pgs_color_1">—</span> Columns</h5>
                <p>It also has been edited to have consistent column headings based on the following schema:</p>
                <div class="table-responsive">
                    <table class="table table-striped table_pgs_auto">
                        <thead class="thead_pgs">
                            <tr>
                                <th>Column Header</th>
                                <th style="width:300px">Field Name</th>
                                <th style="width:130px">Field Requirement</th>
                                <th>Field Description</th>
                            </tr>
                        </thead>
                        <tbody>
                            <tr class="table-tr-hl">
                                <td colspan="4"><i class="fas fa-chevron-right pgs_color"></i> Variant Description:</td>
                            </tr>
                            <tr>
                                <td>rsID</td>
                                <td>dbSNP Accession ID (rsID)</td>
                                <td><span style="cursor:pointer" data-toggle="tooltip" data-placement="right"
                                        data-html="true" title=""
                                        data-original-title="Unless both the <b>chr_name</b> and <b>chr_position</b> columns are absent"><span
                                            class="badge pgs_bg_color_amber">Optional</span> <span
                                            class="fas fa-info-circle pgs_color_1"></span></span></td>
                                <td>The SNP’s rs ID.<br>This column also contains HLA alleles in the standard notation
                                    (e.g. HLA-DQA1*0102) that aren’t always provided with chromosomal positions.</td>
                            </tr>
                            <tr>
                                <td>chr_name</td>
                                <td>Location - Chromosome </td>
                                <td><span style="cursor:pointer" data-toggle="tooltip" data-placement="right"
                                        data-html="true" title=""
                                        data-original-title="May be optional if an <b>rsID</b> for the variant is provided"><span
                                            class="badge pgs_bg_color_red">Required</span> <span
                                            class="fas fa-info-circle pgs_color_1"></span></span></td>
                                <td>Chromosome name/number associated with the variant.</td>
                            </tr>
                            <tr>
                                <td>chr_position</td>
                                <td>Location within the Chromosome</td>
                                <td><span style="cursor:pointer" data-toggle="tooltip" data-placement="right"
                                        data-html="true" title=""
                                        data-original-title="May be optional if an <b>rsID</b> for the variant is provided"><span
                                            class="badge pgs_bg_color_red">Required</span> <span
                                            class="fas fa-info-circle pgs_color_1"></span></span></td>
                                <td>Chromosomal position associated with the variant.</td>
                            </tr>
                            <tr>
                                <td>effect_allele</td>
                                <td>Effect Allele</td>
                                <td><span class="badge pgs_bg_color_red">Required</span></td>
                                <td>The allele that's dosage is counted (e.g. {0, 1, 2}) and multiplied by the variant's
                                    weight (<b>effect_weight</b>) when calculating score. The effect allele is also
                                    known as the 'risk allele'. Note: this does not necessarily need to correspond to
                                    the minor allele/alternative allele.</td>
                            </tr>
                            <tr>
                                <td>other_allele</td>
                                <td>Other allele(s)</td>
                                <td><span style="cursor:pointer" data-toggle="tooltip" data-placement="right"
                                        data-html="true" title=""
                                        data-original-title="Optional, however this column is strongly recommended"><span
                                            class="badge pgs_bg_color_amber">Recommended</span> <span
                                            class="fas fa-info-circle pgs_color_1"></span></span></td>
                                <td>The other allele(s) at the loci. Note: this does not necessarily need to correspond
                                    to the reference allele.</td>
                            </tr>
                            <tr>
                                <td>locus_name</td>
                                <td>Locus Name</td>
                                <td><span class="badge pgs_bg_color_green">Optional</span></td>
                                <td>This is kept in for loci where the variant may be referenced by the gene (APOE e4).
                                    It is also common (usually in smaller PGS) to see the variants named according to
                                    the genes they impact.</td>
                            </tr>
                            <tr>
                                <td>is_haplotype<br>is_diplotype</td>
                                <td>FLAG: Haplotype or Diplotype</td>
                                <td><span class="badge pgs_bg_color_green">Optional</span></td>
                                <td>This is a TRUE/FALSE variable that flags whether the effect allele is a
                                    haplotype/diplotype rather than a single SNP. Constituent SNPs in the haplotype are
                                    semi-colon separated. </td>
                            </tr>
                            <tr>
                                <td>imputation_method</td>
                                <td>Imputation Method</td>
                                <td><span class="badge pgs_bg_color_green">Optional</span></td>
                                <td>This described whether the variant was specifically called with a specific
                                    imputation or variant calling method. This is mostly kept to describe HLA-genotyping
                                    methods (e.g. flag SNP2HLA, HLA*IMP) that gives alleles that are not referenced by
                                    genomic position.</td>
                            </tr>
                            <tr>
                                <td>variant_description</td>
                                <td>Variant Description</td>
                                <td><span class="badge pgs_bg_color_green">Optional</span></td>
                                <td>This field describes any extra information about the variant (e.g. how it is
                                    genotyped or scored) that cannot be captured by the other fields.</td>
                            </tr>
                            <tr>
                                <td>inclusion_criteria</td>
                                <td>Score Inclusion Criteria</td>
                                <td><span class="badge pgs_bg_color_green">Optional</span></td>
                                <td>Explanation of when this variant gets included into the PGS (e.g. if it depends on
                                    the results from other variants).</td>
                            </tr>
                            <tr class="table-tr-hl">
                                <td colspan="4"><i class="fas fa-chevron-right"></i> Weight Information:</td>
                            </tr>
                            <tr>
                                <td>effect_weight</td>
                                <td>Variant Weight</td>
                                <td><span class="badge pgs_bg_color_red">Required</span></td>
                                <td>Value of the effect that is multiplied by the dosage of the effect allele
                                    (<b>effect_allele</b>) when calculating the score. Additional information on how the
                                    effect_weight was derived is in the <b>weight_type</b> field of the header, and
                                    score development method in the metadata downloads.</td>
                            </tr>
                            <tr>
                                <td>is_interaction</td>
                                <td>FLAG: Interaction</td>
                                <td><span class="badge pgs_bg_color_green">Optional</span></td>
                                <td>This is a TRUE/FALSE variable that flags whether the weight should be multiplied
                                    with the dosage of more than one variant. Interactions are demarcated with a
                                    <i>_x_</i> between entries for each of the variants present in the interaction.
                                </td>
                            </tr>
                            <tr>
                                <td>is_dominant</td>
                                <td>FLAG: Dominant Inheritance Model</td>
                                <td><span class="badge pgs_bg_color_green">Optional</span></td>
                                <td>This is a TRUE/FALSE variable that flags whether the weight should be added to the
                                    PGS sum if there is at least 1 copy of the effect allele (e.g. it is a dominant
                                    allele).</td>
                            </tr>
                            <tr>
                                <td>is_recessive</td>
                                <td>FLAG: Recessive Inheritance Model</td>
                                <td><span class="badge pgs_bg_color_green">Optional</span></td>
                                <td>This is a TRUE/FALSE variable that flags whether the weight should be added to the
                                    PGS sum only if there are 2 copies of the effect allele (e.g. it is a recessive
                                    allele).</td>
                            </tr>
                            <tr>
                                <td>dosage_0_weight</td>
                                <td>Effect weight with 0 copy of the effect allele</td>
                                <td><span class="badge pgs_bg_color_green">Optional</span></td>
                                <td rowspan="3">Weights that are specific to different dosages of the
                                    <b>effect_allele</b> (e.g. {0, 1, 2} copies) can also be reported when the the
                                    contribution of the variants to the score is not encoded as additive, dominant, or
                                    recessive. In this case three columns are added corresponding to which variant
                                    weight should be applied for each dosage, where the column name is formated as
                                    <b>dosage_#_weight</b> where the <b>#</b> sign indicates the number of effect_allele
                                    copies.</td>
                            </tr>
                            <tr>
                                <td>dosage_1_weight</td>
                                <td>Effect weight with 1 copy of the effect allele</td>
                                <td><span class="badge pgs_bg_color_green">Optional</span></td>
                            </tr>
                            <tr>
                                <td>dosage_2_weight</td>
                                <td>Effect weight with 2 copies of the effect allele</td>
                                <td><span class="badge pgs_bg_color_green">Optional</span></td>
                            </tr>
                            <tr class="table-tr-hl">
                                <td colspan="4"><i class="fas fa-chevron-right"></i> Other information:</td>
                            </tr>
                            <tr>
                                <td>OR<br>HR</td>
                                <td>Odds Ratio [OR], Hazard Ratio [HR]</td>
                                <td><span class="badge pgs_bg_color_green">Optional</span></td>
                                <td>Author-reported effect sizes can be supplied to the Catalog. If no other
                                    <i>effect_weight</i> is given the weight is calculated using the log(OR) or log(HR).
                                </td>
                            </tr>
                            <tr>
                                <td>allelefrequency_effect</td>
                                <td>Effect Allele Frequency</td>
                                <td><span class="badge pgs_bg_color_green">Optional</span></td>
                                <td>Reported effect allele frequency, if the associated locus is a haplotype then
                                    haplotype frequency will be extracted.</td>
                            </tr>
                            <tr>
                                <td>allelefrequency_effect_<span class="pgs_bracket">Ancestry</span></td>
                                <td>Population-specific effect allele frequency</td>
                                <td><span class="badge pgs_bg_color_green">Optional</span></td>
                                <td>Reported effect allele frequency in a specific population (described by the
                                    authors).</td>
                            </tr>
                        </tbody>
                    </table>
                </div>

                <i class="fas fa-angle-double-right pgs_color_1"></i> <a class="toggle_btn pgs_btn_plus"
                    data-toggle="tooltip" data-placement="right" data-delay="500" id="pgs_data_example" title=""
                    data-original-title="Click to show/hide an example of Scoring Files data">Example of PGS Scoring
                    Files data</a>
                <div class="toggle_list mt-2" id="list_pgs_data_example">
                    <div class="pgs_formatted_block">
                        <pre class="example"><i>Scoring Files header</i>
<span>rsID        chr_name  chr_position  effect_allele  other_allele  effect_weight</span>
rs2843152   1         2245570       G              C             -2.76009e-02
rs35465346  1         22132518      G              A             2.39340e-02
rs28470722  1         38386727      G              A             -1.74935e-02
rs11206510  1         55496039      T              C             2.93005e-02
rs9970807   1         56965664      C              T             4.70027e-02
rs61772626  1         57015668      A              G             -2.71202e-02
rs7528419   1         109817192     A              G             2.91912e-02
rs1277930   1         109822143     A              G             2.60105e-02
rs11102000  1         110298166     G              C             2.45969e-02
rs11810571  1         151762308     G              C             2.09215e-02
rs6689306   1         154395946     G              A             -1.97906e-02
rs72702224  1         154911689     G              A             -2.81310e-02
rs3738591   1         155764808     C              G             4.23731e-02
<span class="pgs_color_2">...</span></pre>
                    </div>
                </div>

                <h5 id="scoring_changes" class="mt-5"><i class="fas fa-copy pgs_color_1 mr-2"></i>Formatted Files <span
                        class="pgs_color_1">—</span> Format changes</h5>
                <ul>
                    <li>
                        <a class="toggle_btn pgs_btn_plus" data-toggle="tooltip" data-placement="right" data-delay="500"
                            id="diff_format_v2_0_vs_v1_0" title=""
                            data-original-title="Click to show/hide the differences between the version 1.0 and 2.0 of the Scoring Files formats"><b>Version
                                2.0</b></a>
                        <div class="toggle_list mt-2 pgs_shaded_block" id="list_diff_format_v2_0_vs_v1_0">
                            <ul>
                                <li>
                                    <b>Header (metadata)</b>
                                    <ul>
                                        <li class="mt-1">
                                            The metadata labels will change from the previous text-based titles to new
                                            field labels that match the outputs of the REST API Score and Publication
                                            endpoints. We will also adopt new formatting by removing spaces (e.g. 'PGS
                                            ID' is changed to 'pgs_id').
                                            <div class="mt-1 pl-1 pgs_comment">This should make the data easier to parse
                                                and more consistent across methods of data access.</div>
                                        </li>
                                        <li class="mt-2">
                                            Addition of the <span class="pgs_color_2"><b>#format_version</b></span>
                                            field.
                                            <div class="mt-1 pl-1 pgs_comment">New versioning to keep track of the
                                                format changes.</div>
                                        </li>
                                        <li class="mt-2">
                                            Addition of the new metadata fields <span
                                                class="pgs_color_2"><b>#trait_mapped</b></span> and <span
                                                class="pgs_color_2"><b>#trait_efo</b></span>.
                                            <div class="mt-1 pl-1 pgs_comment">These fields should make it easier to
                                                group scores based on structured trait terms.</div>
                                        </li>
                                        <li class="mt-2">
                                            Addition of the <span class="pgs_color_2"><b>#weight_type</b></span> field.
                                            <div class="mt-1 pl-1 pgs_comment">This information was previously included
                                                as a column in the scoring file. This lets users know how the <span
                                                    class="pgs_color_dark2">effect_weight</span> column was derived (if
                                                known), otherwise it is labeled as NR (Not Reported).</div>
                                        </li>
                                    </ul>
                                </li>
                                <li class="mt-3">
                                    <b>Scoring File Schema (Column data)</b>
                                    <ul>
                                        <li class="mt-1">
                                            Move redundant information of the column <span
                                                class="pgs_color_1"><b>weight_type</b></span> into the header.
                                            <div class="mt-1 pl-1 pgs_comment">
                                                In all cases the <span class="pgs_color_dark2">weight_type</span> was
                                                identical across all variants in a score.
                                                To remove redundant columns with identical data we moved this
                                                information into a metadata field that is now presented on the website
                                                and API as well as in the scoring file header.
                                            </div>
                                        </li>
                                        <li class="mt-2">
                                            Rename the column <span class="pgs_color_1"><b>reference_allele</b></span>
                                            to <span class="pgs_color_1"><b>other_allele</b></span>.
                                            <div class="mt-1 pl-1 pgs_comment">
                                                We are renaming this column <span
                                                    class="pgs_color_dark2">other_allele</span> to remove the use of the
                                                word "reference". "Reference allele" may have been confused with the
                                                allele included in the reference genome assembly.
                                                The other/non-effect allele may or may not be the reference genome
                                                allele.
                                                Initially we had named it <span
                                                    class="pgs_color_dark2">reference_allele</span> to refer to the
                                                allele that when homozygous has zero (reference) dosage.
                                                We now rename it <span class="pgs_color_dark2">other_allele</span> to
                                                reflect that it is the non-effect allele.
                                                This terminology is consistent with the <a
                                                    href="https://www.ebi.ac.uk/gwas/docs/methods/summary-statistics"
                                                    target="_blank" class="external-link">GWAS summary statistics
                                                    standard format</a> used by the GWAS Catalog.
                                            </div>
                                        </li>
                                    </ul>
                                </li>
                            </ul>

                            <div class="d-flex">
                                <div>
                                    <h5 class="mt-4" style="text-align:center">Format version <b>2.0</b></h5>
                                    <div class="pgs_formatted_block">
                                        <pre><span>###PGS CATALOG SCORING FILE - see ...</span>
<span class="pgs_color_2"><b>#format_version</b></span>=<i>Version of the scoring file format, e.g. '2.0'</i>
<span>##POLYGENIC SCORE (PGS) INFORMATION</span>
<b>#pgs_id</b>=<i>PGS identifier, e.g. 'PGS000001'</i>
<b>#pgs_name</b>=<i>PGS name, e.g. 'PRS77_BC' - optional</i>
<b>#trait_reported</b>=<i>trait, e.g. 'Breast Cancer'</i>
<span class="pgs_color_2"><b>#trait_mapped</b></span>=<i>Ontology trait name, e.g. 'breast carcinoma'</i>
<span class="pgs_color_2"><b>#trait_efo</b></span>=<i>Ontology trait ID (EFO), e.g. 'EFO_0000305'</i>
<b>#genome_build</b>=<i>Genome build/assembly, e.g. 'GRCh38'</i>
<b>#variants_number</b>=<i>Number of variants listed in the PGS</i>
<span class="pgs_color_2"><b>#weight_type</b></span>=<i>Variant weight type, e.g. 'beta', 'OR/HR' (default 'NR')</i>
<span>##SOURCE INFORMATION</span>
<b>#pgp_id</b>=<i>PGS publication identifier, e.g. 'PGP000001'</i>
<b>#citation</b>=<i>Information about the publication</i>
<b>#license</b>=<i>License and terms of PGS use/distribution ...</i>
<span class="scoring_col">rsID</span><span class="scoring_col">chr_name</span><span class="scoring_col">chr_position</span><span class="scoring_col">effect_allele</span><span class="scoring_col pgs_color_1">other_allele</span><span class="pgs_color_2">...</span></pre>
                                    </div>
                                </div>
                                <div>
                                    <h5 class="mt-4" style="text-align:center">Schema version <b>1.0</b></h5>
                                    <div class="pgs_formatted_block">
                                        <pre><span>### PGS CATALOG SCORING FILE - see ...</span>

<span>## POLYGENIC SCORE (PGS) INFORMATION</span>
<b># PGS ID</b> = <i>PGS identifier, e.g. 'PGS000001'</i>
<b># PGS Name</b> = <i>PGS name, e.g. 'PRS77_BC' - optional</i>
<b># Reported Trait</b> = <i>trait, e.g. 'Breast Cancer'</i>


<b># Original Genome Build</b> = <i>Genome build/assembly, e.g. 'GRCh38'</i>
<b># Number of Variants</b> = <i>Number of variants listed in the PGS</i>

<span>## SOURCE INFORMATION</span>
<b># PGP ID</b> = <i>PGS publication identifier, e.g. 'PGP000001'</i>
<b># Citation</b> = <i>Information about the publication</i>
<b># LICENSE</b> = <i>License and terms of PGS use/distribution ...</i>
<span class="scoring_col">rsID</span><span class="scoring_col">chr_name</span><span class="scoring_col">chr_position</span><span class="scoring_col">effect_allele</span><span class="scoring_col pgs_color_1">reference_allele</span><span class="pgs_color_2">...</span></pre>
                                    </div>
                                </div>
                            </div>
                            <div>
                                <ul>
                                    <li><span class="pgs_color_2">New header information</span></li>
                                    <li><span class="pgs_color_1">Column name change</span></li>
                                </ul>
                            </div>
                        </div>
                    </li>
                </ul>


                <h5 id="scoring_previous" class="mt-5"><i class="far fa-copy pgs_color_1 mr-2"></i>Formatted Files <span
                        class="pgs_color_1">—</span> Previous formats</h5>
                <ul>
                    <li>
                        <a class="toggle_btn pgs_btn_plus" data-toggle="tooltip" data-placement="right" data-delay="500"
                            id="format_v1_0" title=""
                            data-original-title="Click to show/hide the description of the Scoring Files format 1.0"><b>Version
                                1.0</b> (up to December 14, 2021)</a>
                        <div class="toggle_list mt-2 pgs_shaded_block" id="list_format_v1_0">
                            <h5 class="mt-0">Header</h5>
                            <p>Here is a description of the PGS Scoring Files header:</p>

                            <div class="pgs_formatted_block">
                                <pre><span>### PGS CATALOG SCORING FILE - see https://www.pgscatalog.org/downloads/#dl_ftp_scoring for additional information</span>
<span>## POLYGENIC SCORE (PGS) INFORMATION</span>
<b># PGS ID</b> = <i>PGS identifier, e.g. 'PGS000001'</i>
<b># PGS Name</b> = <i>PGS name, e.g. 'PRS77_BC' - optional</i>
<b># Reported Trait</b> = <i>trait, e.g. 'Breast Cancer'</i>
<b># Original Genome Build</b> = <i>Genome build/assembly, e.g. 'GRCh38'</i>
<b># Number of Variants</b> = <i>Number of variants listed in the PGS</i>
<span>## SOURCE INFORMATION</span>
<b># PGP ID</b> = <i>PGS publication identifier, e.g. 'PGP000001'</i>
<b># Citation</b> = <i>Information about the publication</i>
<b># LICENSE</b> = <i>License and terms of PGS use/distribution - refers to the EMBL-EBI Terms of Use by default</i>
<span class="scoring_col">rsID</span><span class="scoring_col">chr_name</span><span class="scoring_col">chr_position</span><span class="scoring_col">effect_allele</span><span class="scoring_col">reference_allele</span><span class="pgs_color_2">...</span></pre>
                            </div>

                            <i class="fas fa-angle-double-right pgs_color_1"></i> <a class="toggle_btn pgs_btn_plus"
                                data-toggle="tooltip" data-placement="right" data-delay="500"
                                id="pgs_header_example_v1_0" title=""
                                data-original-title="Click to show/hide an example of Scoring Files header">Example of
                                PGS Scoring Files header</a>
                            <div class="toggle_list mt-2" id="list_pgs_header_example_v1_0">
                                <div class="pgs_formatted_block">
                                    <pre
                                        class="example"><span>### PGS CATALOG SCORING FILE - see https://www.pgscatalog.org/downloads/#dl_ftp_scoring for additional information</span>
<span>## POLYGENIC SCORE (PGS) INFORMATION</span>
<b># PGS ID</b> = PGS000348
<b># PGS Name</b> = PRS_PrCa
<b># Reported Trait</b> = Prostate cancer
<b># Original Genome Build</b> = GRCh37
<b># Number of Variants</b> = 72
<span>## SOURCE INFORMATION</span>
<b># PGP ID</b> = PGP000113
<b># Citation</b> = Black M et al. Prostate (2020). doi:10.1002/pros.24058
<b># LICENSE</b> = Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International (CC BY-NC-ND 4.0). © 2020 Ambry Genetics.
<span class="scoring_col">rsID</span><span class="scoring_col">chr_name</span><span class="scoring_col">chr_position</span><span class="scoring_col">effect_allele</span><span class="scoring_col">reference_allele</span><span class="scoring_col">effect_weight</span><span class="pgs_color_2">...</span></pre>
                                </div>
                            </div>

                            <h5 class="mt-5">Columns</h5>
                            <p>It also has been edited to have consistent column headings based on the following schema:
                            </p>
                            <div class="table-responsive">
                                <table class="table table-striped table_pgs_auto">
                                    <thead class="thead_pgs">
                                        <tr>
                                            <th>Column Header</th>
                                            <th>Field Name</th>
                                            <th>Field Description</th>
                                            <th>Field Requirement</th>
                                        </tr>
                                    </thead>
                                    <tbody>
                                        <tr>
                                            <td>rsID</td>
                                            <td>dbSNP Accession ID (rsID) </td>
                                            <td>The SNP’s rs ID</td>
                                            <td><span class="pgs_color_amber">Optional</span> - unless both the
                                                <i>chr_name</i> and <i>chr_position</i> columns are absent. This column
                                                also contains HLA alleles in the standard notation (e.g. HLA-DQA1*0102)
                                                that aren’t always provided with chromosomal positions.</td>
                                        </tr>
                                        <tr>
                                            <td>chr_name</td>
                                            <td>Location - Chromosome </td>
                                            <td>Chromosome name/number associated with the variant</td>
                                            <td><span class="pgs_color_red">Required</span> - may be optional if an
                                                <i>rsID</i> for the variant is provided</td>
                                        </tr>
                                        <tr>
                                            <td>chr_position</td>
                                            <td>Location within the Chromosome</td>
                                            <td>Chromosomal position associated with the variant</td>
                                            <td><span class="pgs_color_red">Required</span> - may be optional if an
                                                <i>rsID</i> for the variant is provided</td>
                                        </tr>
                                        <tr>
                                            <td>effect_allele</td>
                                            <td>Effect Allele</td>
                                            <td>The allele that's dosage is counted (e.g. {0, 1, 2}) and multiplied by
                                                the variant's weight ('effect_weight') when calculating score. The
                                                effect allele is also known as the 'risk allele'.</td>
                                            <td><span class="pgs_color_red">Required</span></td>
                                        </tr>
                                        <tr>
                                            <td>reference_allele</td>
                                            <td>Reference Allele</td>
                                            <td>The other allele(s) at the loci</td>
                                            <td><span class="pgs_color_green">Optional</span> - but strongly recommended
                                            </td>
                                        </tr>
                                        <tr>
                                            <td>effect_weight</td>
                                            <td>Variant Weight</td>
                                            <td>Value of the effect that is multiplied by the dosage of the effect
                                                allele ('effect_allele') when calculating the score.</td>
                                            <td><span class="pgs_color_red">Required</span></td>
                                        </tr>
                                        <tr>
                                            <td>locus_name</td>
                                            <td>Locus Name</td>
                                            <td>This is kept in for loci where the variant may be referenced by the gene
                                                (APOE e4). It is also common (usually in smaller PGS) to see the
                                                variants named according to the genes they impact.</td>
                                            <td><span class="pgs_color_green">Optional</span></td>
                                        </tr>
                                        <tr>
                                            <td>weight_type</td>
                                            <td>Type of Weight</td>
                                            <td>Whether the author supplied Variant Weight is a: <b>beta</b> (effect
                                                size), or something like an <b>OR/HR</b> (odds/hazard ratio)</td>
                                            <td><span class="pgs_color_green">Optional</span></td>
                                        </tr>
                                        <tr>
                                            <td>allelefrequency_effect</td>
                                            <td>Effect Allele Frequency</td>
                                            <td>Reported effect allele frequency, if the associated locus is a haplotype
                                                then haplotype frequency will be extracted.</td>
                                            <td><span class="pgs_color_green">Optional</span></td>
                                        </tr>
                                        <tr>
                                            <td>is_interaction</td>
                                            <td>FLAG: Interaction</td>
                                            <td>This is a TRUE/FALSE variable that flags whether the weight should be
                                                multiplied with the dosage of more than one variant. Interactions are
                                                demarcated with a <i>_x_</i> between entries for each of the variants
                                                present in the interaction. </td>
                                            <td><span class="pgs_color_green">Optional</span></td>
                                        </tr>
                                        <tr>
                                            <td>is_dominant</td>
                                            <td>FLAG: Dominant Inheritance Model</td>
                                            <td>This is a TRUE/FALSE variable that flags whether the weight should be
                                                added to the PGS sum if there is at least 1 copy of the effect allele
                                                (e.g. it is a dominant allele).</td>
                                            <td><span class="pgs_color_green">Optional</span></td>
                                        </tr>
                                        <tr>
                                            <td>is_recessive</td>
                                            <td>FLAG: Recessive Inheritance Model</td>
                                            <td>This is a TRUE/FALSE variable that flags whether the weight should be
                                                added to the PGS sum only if there are 2 copies of the effect allele
                                                (e.g. it is a recessive allele).</td>
                                            <td><span class="pgs_color_green">Optional</span></td>
                                        </tr>
                                        <tr>
                                            <td>is_haplotype<br>is_diplotype</td>
                                            <td>FLAG: Haplotype or Diplotype</td>
                                            <td>This is a TRUE/FALSE variable that flags whether the effect allele is a
                                                haplotype/diplotype rather than a single SNP. Constituent SNPs in the
                                                haplotype are semi-colon separated. </td>
                                            <td><span class="pgs_color_green">Optional</span></td>
                                        </tr>
                                        <tr>
                                            <td>imputation_method</td>
                                            <td>Imputation Method</td>
                                            <td>This described whether the variant was specifically called with a
                                                specific imputation or variant calling method. This is mostly kept to
                                                describe HLA-genotyping methods (e.g. flag SNP2HLA, HLA*IMP) that gives
                                                alleles that are not referenced by genomic position.</td>
                                            <td><span class="pgs_color_green">Optional</span></td>
                                        </tr>
                                        <tr>
                                            <td>variant_description</td>
                                            <td>Variant Description</td>
                                            <td>This field describes any extra information about the variant (e.g. how
                                                it is genotyped or scored) that cannot be captured by the other fields.
                                            </td>
                                            <td><span class="pgs_color_green">Optional</span></td>
                                        </tr>
                                        <tr>
                                            <td>inclusion_criteria</td>
                                            <td>Score Inclusion Criteria</td>
                                            <td>Explanation of when this variant gets included into the PGS (e.g. if it
                                                depends on the results from other variants).</td>
                                            <td><span class="pgs_color_green">Optional</span></td>
                                        </tr>
                                        <tr class="table-tr-hl">
                                            <td colspan="4"><i class="fas fa-chevron-right"></i> Extra columns:</td>
                                        </tr>
                                        <tr>
                                            <td>OR<br>HR</td>
                                            <td>Odds Ratio [OR], Hazard Ratio [HR]</td>
                                            <td>Author-reported effect sizes can be supplied to the Catalog. If no other
                                                <i>effect_weight</i> is given the weight is calculated using the log(OR)
                                                or log(HR).</td>
                                            <td><span class="pgs_color_green">Optional</span></td>
                                        </tr>
                                        <tr>
                                            <td>allelefrequency_effect_<span class="pgs_bracket">Ancestry</span></td>
                                            <td>Population-specific effect allele frequency</td>
                                            <td>Reported effect allele frequency in a specific population (described by
                                                the authors).</td>
                                            <td><span class="pgs_color_green">Optional</span></td>
                                        </tr>
                                    </tbody>
                                </table>
                            </div>

                            <i class="fas fa-angle-double-right pgs_color_1"></i> <a class="toggle_btn pgs_btn_plus"
                                data-toggle="tooltip" data-placement="right" data-delay="500" id="pgs_data_example_v1_0"
                                title=""
                                data-original-title="Click to show/hide an example of Scoring Files data">Example of PGS
                                Scoring Files data</a>
                            <div class="toggle_list mt-2" id="list_pgs_data_example_v1_0">
                                <div class="pgs_formatted_block">
                                    <pre class="example"><i>Scoring Files header</i>
<span>rsID        chr_name  chr_position  effect_allele  reference_allele  effect_weight</span>
rs2843152   1         2245570       G              C                 -2.76009e-02
rs35465346  1         22132518      G              A                  2.39340e-02
rs28470722  1         38386727      G              A                 -1.74935e-02
rs11206510  1         55496039      T              C                  2.93005e-02
rs9970807   1         56965664      C              T                  4.70027e-02
rs61772626  1         57015668      A              G                 -2.71202e-02
rs7528419   1         109817192     A              G                  2.91912e-02
rs1277930   1         109822143     A              G                  2.60105e-02
rs11102000  1         110298166     G              C                  2.45969e-02
rs11810571  1         151762308     G              C                  2.09215e-02
rs6689306   1         154395946     G              A                 -1.97906e-02
rs72702224  1         154911689     G              A                 -2.81310e-02
rs3738591   1         155764808     C              G                  4.23731e-02
<span class="pgs_color_2">...</span></pre>
                                </div>
                            </div>
                        </div>
                    </li>
                </ul>
            </div>

        </section>

        <section class="mt-5">
            <div class="d-inline-flex">
                <h4 class="mr-3" id="dl_ftp_scoring_hm_pos">Harmonized Files</h4>
                <div class="mt-4">
                    <div class="mt-2 badge badge-pgs">Format: 2.0</div>
                </div>
            </div>

            <div class="ml-3">

                <div class="d-flex mb-3">
                    <div class="pgs_shaded_block">
                        <i class="fas fa-file-signature pgs_color_1"></i> <a href="#hm_pos_fn">File name</a><span
                            class="pl-2 pr-2">—</span><i class="fas fa-hashtag pgs_color_1"></i> <a
                            href="#hm_pos_header">Header</a><span class="pl-2 pr-2">—</span><i
                            class="fas fa-th pgs_color_1"></i> <a href="#hm_pos_columns">Additional Columns</a>
                    </div>
                </div>

                <p>
                    PGS Scoring Files in the Catalog are currently provided in a consistent format with standardized
                    column names and data types, along with information about the genome build given by authors.
                    The variant-level information in PGS is often heterogeneously described and may lack
                    chromosome/position information, contain a mix of positions and/or rsIDs, or be mapped to a genome
                    build different from your sample genotypes.
                    To make PGS easier to apply we have created a new set of files that contain <b>additional
                        columns</b> with harmonized variant information (chromosome name and base pair position) and
                    variant identifiers (updated rsID), in commonly used genome builds (GRCh37/hg19 and GRCh38/hg38) to
                    make variant matching and PGS calculation easier.
                </p>
                <span>
                    The generation of these harmonized files is done by using the <a
                        href="https://github.com/PGScatalog/pgs-harmonizer" target="_blank"
                        class="external-link">pgs-harmonizer</a> tool. It is based on the <a
                        href="https://github.com/EBISPOT/gwas-sumstats-harmoniser" target="_blank"
                        class="external-link">Open Targets and GWAS Catalog Summary Statistics harmonizer pipelines</a>.
                    To harmonize the variant positions the pgs-harmonizer performs the following tasks:
                </span>
                <ul>
                    <li class="mt-2">
                        <b>Mapping rsIDs to chromosomal positions</b>: we use <a href="https://www.ensembl.org"
                            target="_blank" class="external-link">Ensembl</a> (VCF files and REST APIs) on GRCh37 and
                        GRCh38. We use Ensembl version <b>105</b>.
                    </li>
                    <li class="mt-2">
                        <b>Liftover - mapping chromosomal positions across builds</b> (only when generating a Scoring
                        file on a different genome build): we use the UCSC liftover tools via the Python library <a
                            href="https://github.com/konstantint/pyliftover" target="_blank"
                            class="external-link">pyliftover</a>.
                    </li>
                </ul>
                <p>
                    The resultant files create <a href="#hm_pos_columns"><i class="fas fa-th pgs_color_1 mr-1"></i>new
                        columns</a>, indicating the source of the variant annotation (<b>hm_source</b>), as well as
                    consistently annotated chromosome (<b>hm_chr</b>) / position (<b>hm_pos</b>), and rsID
                    (<b>hm_rsID</b>) which can be used to match variants in your dataset along with the alleles
                    (<b>effect_allele</b>, and <b>other_allele</b>).
                </p>

                <div class="d-flex mb-3">
                    <div class="d-flex mb-2 pgs_note pgs_note_2">
                        <div>
                            <span>Warning: </span>Complex variants in the human leukocyte antigen (HLA) region
                            (alleles/haplotypes/diplotypes) and larger copy number variants (CNVs) without explicit
                            positions will have missing harmonized positions as they are not indexed in the ENSEMBL
                            VCFs.
                            For these variants we suggest using the author-reported annotations, specifically the
                            <b>effect_allele</b> column, for variant matching in absence of positional information.
                        </div>
                    </div>
                </div>

                <span>Harmonized scoring files can be accessed via our FTP, in a separate directory for each
                    score:</span>
                <div class="pgs_formatted_block">
                    <pre>ftp://ftp.ebi.ac.uk/pub/databases/spot/pgs/<b>scores</b>/<b>PGS######</b>/<b>ScoringFiles</b>/<b>Harmonized</b>/</pre>
                </div>

                <div class="d-flex mb-1">
                    <div class="d-flex mb-2 pgs_note pgs_note_1">
                        <div>
                            <span>Note: </span>The harmonized scoring file URLs can also be found for any Score result
                            within the <a href="/rest">REST API</a>.
                        </div>
                    </div>
                </div>

                <h5 id="hm_pos_fn" class="mt-5"><i class="fas fa-file-signature pgs_color_1 mr-2"></i>Harmonized Files
                    <span class="pgs_color_1">—</span> File name</h5>
                The file name is composed of 3 parts, separated by underscores (<code>_</code>):
                <ol>
                    <li><b>PGS ID</b>, e.g. 'PGS000001'</li>
                    <li><b>Type of harmonized file</b>, e.g. 'hmPOS'</li>
                    <li><b>Genome build</b>, either 'GRCh37' or ‘GRCh38’</li>
                </ol>
                <p>For instance: <b>PGS000001</b><code>_</code><b>hmPOS</b><code>_</code><b>GRCh37</b>.txt.gz</p>


                <h5 id="hm_pos_header" class="mt-5"><i class="fas fa-hashtag pgs_color_1 mr-2"></i>Harmonized Files
                    <span class="pgs_color_1">—</span> Header</h5>

                <div class="d-flex mb-1">
                    <div class="d-flex mb-2 pgs_note pgs_note_1">
                        <div>
                            <span>Note: </span>The first part of the header (before the line
                            <code>##HARMONIZATION DETAILS</code>) is a copy-paste of the Scoring file header.
                        </div>
                    </div>
                </div>
                <p>Here is a description of the PGS Harmonized Scoring Files header:</p>

                <div class="pgs_formatted_block">
                    <pre><span>###PGS CATALOG SCORING FILE - see https://www.pgscatalog.org/downloads/#dl_ftp_scoring for additional information</span>
<b>#format_version</b>=<i>Version of the scoring file format, e.g. '2.0'</i>
<span>##POLYGENIC SCORE (PGS) INFORMATION</span>
<b>#pgs_id</b>=<i>PGS identifier, e.g. 'PGS000001'</i>
<span>...</span>
<b>#license</b>=<i>License and terms of PGS use/distribution - refers to the EMBL-EBI Terms of Use by default</i>
<span>##HARMONIZATION DETAILS</span>
<b>#HmPOS_build</b>=<i>Genome build of the harmonized file, e.g. 'GRCh38'</i>
<b>#HmPOS_date</b>=<i>Date of the harmonized file creation, e.g. '2022-05-26'</i>
<b>#HmPOS_match_chr</b>=<i>Number of entries matching and not matching the given chromosome, e.g. {"True": 5210, "False": 8}</i>
<b>#HmPOS_match_pos</b>=<i>Number of entries matching and not matching the given position, e.g. {"True": 5210, "False": 8}</i>
<span class="scoring_col">rsID</span><span class="scoring_col">...</span><span class="scoring_col">hm_source</span><span class="scoring_col">hm_rsID</span><span class="scoring_col">hm_chr</span><span class="scoring_col">hm_pos</span><span class="scoring_col">hm_inferOtherAllele</span><span class="scoring_col">hm_match_chr</span><span class="scoring_col">hm_match_pos</span></pre>
                </div>

                <i class="fas fa-angle-double-right pgs_color_1"></i> <a class="toggle_btn pgs_btn_plus"
                    data-toggle="tooltip" data-placement="right" data-delay="500" id="pgs_hm_header_example" title=""
                    data-original-title="Click to show/hide an example of Scoring Files header">Example of PGS Scoring
                    Files header</a>
                <div class="toggle_list mt-2" id="list_pgs_hm_header_example">
                    <div class="pgs_formatted_block">
                        <pre
                            class="example"><span>###PGS CATALOG SCORING FILE - see https://www.pgscatalog.org/downloads/#dl_ftp_scoring for additional information</span>
<b>#format_version</b>=2.0
<span>##POLYGENIC SCORE (PGS) INFORMATION</span>
<b>#pgs_id</b>=PGS000348
<span>...</span>
<b>#license</b>=Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International (CC BY-NC-ND 4.0). © 2020 Ambry Genetics.
<span>##HARMONIZATION DETAILS</span>
<b>#HmPOS_build</b>=GRCh37
<b>#HmPOS_date</b>=2022-07-26
<b>#HmPOS_match_chr</b>={"True": 72, "False":0}
<b>#HmPOS_match_pos</b>={"True": 72, "False":0}
<span class="scoring_col">rsID</span><span class="scoring_col">...</span><span class="scoring_col">hm_source</span><span class="scoring_col">hm_rsID</span><span class="scoring_col">hm_chr</span><span class="scoring_col">hm_pos</span><span class="scoring_col">hm_inferOtherAllele</span><span class="scoring_col">hm_match_chr</span><span class="scoring_col">hm_match_pos</span></pre>
                    </div>
                </div>

                <h5 id="hm_pos_columns" class="mt-5"><i class="fas fa-th pgs_color_1 mr-2"></i>Harmonized Files <span
                        class="pgs_color_1">—</span> Additional Columns</h5>

                <p>The formatted scoring file (in the original genome build) has the following additional columns
                    describing the variants in the <b>specified genome build</b> for each HmPOS file:</p>

                <div class="table-responsive">
                    <table class="table table-striped table_pgs_auto">
                        <thead class="thead_pgs">
                            <tr>
                                <th>Additional Column Header</th>
                                <th style="width:300px">Field Name</th>
                                <th>Field Description</th>
                            </tr>
                        </thead>
                        <tbody>
                            <tr>
                                <td>hm_source</td>
                                <td>Provider of the harmonized variant information</td>
                                <td>Data source of the variant position. Options include: ENSEMBL, liftover,
                                    author-reported <i>(if being harmonized to the same build)</i>.</td>
                            </tr>
                            <tr>
                                <td>hm_rsID</td>
                                <td>Harmonized rsID</td>
                                <td>Current rsID. Differences between this column and the author-reported column (rsID)
                                    indicate variant merges and annotation updates from dbSNP.</td>
                            </tr>
                            <tr>
                                <td>hm_chr</td>
                                <td>Harmonized chromosome name</td>
                                <td>Chromosome that the harmonized variant is present on, preferring matches to
                                    chromosomes over patches present in later builds.</td>
                            </tr>
                            <tr>
                                <td>hm_pos</td>
                                <td>Harmonized chromosome position</td>
                                <td>Chromosomal position (base pair location) where the variant is located, preferring
                                    matches to chromosomes over patches present in later builds.</td>
                            </tr>
                            <tr>
                                <td>hm_inferOtherAllele</td>
                                <td>Harmonized other alleles</td>
                                <td>If only the <b>effect_allele</b> is given we attempt to infer the non-effect/other
                                    allele(s) using Ensembl/dbSNP alleles.</td>
                            </tr>
                            <tr>
                                <td>hm_match_chr</td>
                                <td>FLAG: matching chromosome name</td>
                                <td><i>Used for QC</i>. Only provided if the scoring file is being harmonized to the
                                    same genome build, and where the chromosome name is provided in the column
                                    <b>chr_name</b>.</td>
                            </tr>
                            <tr>
                                <td>hm_match_pos</td>
                                <td>FLAG: matching chromosome position</td>
                                <td><i>Used for QC</i>. Only provided if the scoring file is being harmonized to the
                                    same genome build, and where the chromosome name is provided in the column
                                    <b>chr_position</b>.</td>
                            </tr>
                        </tbody>
                    </table>
                </div>
                <i class="fas fa-angle-double-right pgs_color_1"></i> <a class="toggle_btn pgs_btn_plus"
                    data-toggle="tooltip" data-placement="right" data-delay="500" id="pgs_hm_content_example" title=""
                    data-original-title="Click to show/hide an example of Scoring Files header">Example of PGS
                    Harmonized File (<span class="pgs_color_facet_2">GRCh37</span> file harmonized on <span
                        class="pgs_color_facet_3">GRCh38</span>)</a>
                <div class="toggle_list mt-2" id="list_pgs_hm_content_example">
                    <div class="pgs_formatted_block">
                        <pre class="example"><span>###PGS CATALOG SCORING FILE - see https://www.pgscatalog.org/downloads/#dl_ftp_scoring for additional information</span>
<b>#format_version</b>=2.0
<span>##POLYGENIC SCORE (PGS) INFORMATION</span>
<b>#pgs_id</b>=PGS000116
<span>...</span>
<b>#genome_build</b>=<span class="pgs_color_2">GRCh37</span>
<span>...</span>
<span>##HARMONIZATION DETAILS</span>
<b>#HmPOS_build</b>=<span class="pgs_color_facet_3">GRCh38</span>
<span>...</span>
<span class="pgs_color_facet_2">rsID</span>         <span class="pgs_color_facet_2">chr_name</span>  <span class="pgs_color_facet_2">chr_position</span>  <span class="scoring_col pr-0">effect_allele</span>  <span class="scoring_col pr-0">other_allele</span>  <span class="scoring_col pr-0">effect_weight</span>  <span class="scoring_col pr-0">hm_source</span>  <span class="pgs_color_facet_3">hm_rsID</span>      <span class="pgs_color_facet_3">hm_chr</span>  <span class="pgs_color_facet_3">hm_pos</span>
rs1921       1          949608       A              G             -0.003965      ENSEMBL    rs1921       1       1014228
rs2710887    1          986443       T              C             -0.000846      ENSEMBL    rs2710887    1       1051063
rs11260596   1         1002434       T              C              0.000789      ENSEMBL    rs11260596   1       1067054
rs113355263  1         1069535       A              G             -0.001627      ENSEMBL    rs113355263  1       1134155
rs11260539   1         1109903       T              C              0.000170      ENSEMBL    rs11260539   1       1174523
...</pre>
                    </div>
                </div>
            </div>

        </section>


    </div>


    <footer class="footer font-small pt-3">

        <div class="footer_first_row">
            <div>Supported by:</div>
            <div>
                <a class="pgs_no_icon_link" href="https://www.phpc.cam.ac.uk/" target="_blank"><img
                        class="img-fluid pgs_small_icons" src="/static/catalog/imgs/UofCambridge_small.png"
                        alt="University of Cambridge"></a>
            </div>
            <div>
                <a class="pgs_no_icon_link" href="https://www.ebi.ac.uk" style="link:none" target="_blank"><img
                        class="img-fluid pgs_small_icons" src="/static/catalog/imgs/EMBL_EBI_small.png"
                        alt="EMBL-EBI"></a>
            </div>
            <div>
                <a class="pgs_no_icon_link"
                    href="https://www.hdruk.ac.uk/about-us/our-locations/health-data-research-uk-hdr-uk-cambridge/"
                    target="_blank"><img class="img-fluid pgs_small_icons" src="/static/catalog/imgs/HDR-UK_small.png"
                        alt="HDR-UK (Cambridge)"></a>
            </div>
            <div>
                <a class="pgs_no_icon_link" href="https://baker.edu.au" target="_blank"><img
                        class="img-fluid pgs_small_icons" src="/static/catalog/imgs/Baker_Logo_small.png"
                        alt="Baker Heart and Diabetes Institute"></a>
            </div>
            <div>
                <a class="pgs_no_icon_link" href="https://www.genome.gov/" target="_blank"><img
                        class="img-fluid pgs_small_icons" src="/static/catalog/imgs/NHGRI_Logo_small.png"
                        alt="National Human Genome Research Institute"></a>
            </div>
            <div>
                <div class="pgs_social">
                    <a class="pgs_no_icon_link fas fa-envelope" title="Contact us" href="mailto:pgs-info@ebi.ac.uk"
                        target="_blank"></a>
                    <a class="pgs_no_icon_link fab fa-twitter" title="Twitter @PGSCatalog"
                        href="https://www.twitter.com/pgscatalog" target="_blank"></a>
                </div>
            </div>
        </div>

        <!-- Copyright -->
        <div class="footer-copyright text-center py-3">
            <span>Copyright © EMBL-EBI 2024</span>
            <span><a class="pgs_no_icon_link" href="https://www.ebi.ac.uk" target="_blank">EMBL-EBI</a> is an Outstation
                of the <a class="pgs_no_icon_link" href="https://www.embl.org/" target="_blank">European Molecular
                    Biology Laboratory</a></span>
            <span><a class="pgs_no_icon_link" href="https://www.ebi.ac.uk/about/privacy"
                    target="_blank">Privacy</a></span>
            <span><a class="pgs_no_icon_link" href="https://www.ebi.ac.uk/about/cookies"
                    target="_blank">Cookies</a></span>
            <span><a class="pgs_no_icon_link" href="https://www.ebi.ac.uk/about/terms-of-use" target="_blank">Terms of
                    use</a></span>
            <span><a class="pgs_no_icon_link" href="https://www.ebi.ac.uk/long-term-data-preservation"
                    target="_blank">Data Preservation Statement</a></span>
        </div>
        <!-- Copyright -->
    </footer>


    <!-- Global site tag (gtag.js) - Google Analytics -->
    <script async="" src="https://www.googletagmanager.com/gtag/js?id=G-9B3G71V98V"></script>
    <script>
        window.dataLayer = window.dataLayer || [];
        function gtag() { dataLayer.push(arguments); }
        gtag('js', new Date());
        gtag('config', 'G-9B3G71V98V');
    </script>





    <ul id="ui-id-1" tabindex="0" class="ui-menu ui-widget ui-widget-content ui-autocomplete ui-front"
        style="display: none;" unselectable="on"></ul>
    <div role="status" aria-live="assertive" aria-relevant="additions" class="ui-helper-hidden-accessible"></div>
    <ul id="ui-id-2" tabindex="0" class="ui-menu ui-widget ui-widget-content ui-autocomplete ui-front"
        style="display: none;" unselectable="on"></ul>
    <div role="status" aria-live="assertive" aria-relevant="additions" class="ui-helper-hidden-accessible"></div>
    <div id="data-protection-banner" class="data-protection-banner"
        style="position: fixed; background: rgb(17, 17, 17); width: 100%; padding: 0.75rem 1%; left: 0px; bottom: 0px; color: rgb(238, 238, 238); z-index: 10; display: block;">
        <div class="row">
            <div class="columns medium-8 large-9 white-color">This website requires cookies, and the limited processing
                of your personal data in order to function. By using the site you are agreeing to this as outlined in
                our <a target="_blank"
                    href="https://www.ebi.ac.uk/data-protection/privacy-notice/embl-ebi-public-website">Privacy
                    Notice</a> and <a target="_blank" href="https://www.ebi.ac.uk/about/terms-of-use">Terms of Use</a>.
            </div>
            <div class="columns medium-4 large-3 text-right white-color"><a id="data-protection-agree" class="">I agree,
                    dismiss this banner</a></div>
        </div>
    </div>
</body>

</html>